src/services/smartSelection.ts

```diff
@@ -184,7 +195,7 @@ namespace ts.SmartSelectionRange {
         if (isPropertySignature(node)) {
             const children = groupChildren(node.getChildren(), child =>
                 child === node.name || contains(node.modifiers, child));
-            return splitChildren(children, ({ kind }) => kind === SyntaxKind.ColonToken);
+            return splitChildren(children, ({ kind }) => kind === SyntaxKind.ColonToken || kind === SyntaxKind.JSDoc);

```

Comment:
This doesn’t work too well when the property signature has modifiers. Try adding `readonly` before `status` in smartSelection_JSDocTags11.ts and notice how the whole line appears at once since we were no longer able to pivot on the colon token. There needs to be another level in the hierarchy rather than just shifting the pivot point. It also seems like that should be happening for more than just property signatures. In other words, if we have a `children` like

```
[
  JSDoc,
  X,
  Y,
  Z
]
```

maybe it always makes sense to restructure it like

```
[
  JSDoc,
  SyntaxList([
    X,
    Y,
    Z
  ])
]
```

Reply:
Yes. You're right. I did as you suggested and it's working. Also, updated the test case to include the modifier (and a new cursor position at the start of it).